### PR TITLE
[SPMD] Move some functions to IterGraphModule.setup()

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99076
* #99075
* #99069

Since users will have to call these steps before calling `setup()`, moving these steps to `setup()` can reduce the API usage complexity.

Differential Revision: [D44973726](https://our.internmc.facebook.com/intern/diff/D44973726/)

### CODE REVIEW

File: torch/distributed/_spmd/gm_transformation.py [hunk 1]
```diff
@@ -33,12 +32,12 @@ def __call__(self, gm: fx.GraphModule) -> Callable:
...
 
-        iter_gm = IterGraphModule(gm)
+        iter_gm = IterGraphModule(gm, enable_inductor=self.enable_inductor)
         if self.enable_graph_optimization:
             comm_fusion_with_concat(iter_gm, 100)
             schedule_comm_wait(iter_gm)
             remove_copy_from_optimizer(iter_gm)
-        iter_gm.freeze_cross_iter_movement()
+        # Must be called after we are not going to move the graphs
         iter_gm.setup(self.num_iters)
```
Comment (lessw2020) [inline torch/distributed/_spmd/gm_transformation.py:41]: future idea not related to this PR directly - seems like 'finalize' would be clearer name than setup here (setup implies first step, vs this is last step before locking down graphs and also this means having multiple setup references - could be confused with the 'setup' graph as an entity vs this as a function).

**Review by lessw2020 — APPROVED** [review]
looks good.

File: torch/distributed/_spmd/gm_transformation.py (hunk 1 above)
Comment (fegin) [reply torch/distributed/_spmd/gm_transformation.py:41]: Thanks for the suggestion. Right now we are unable to change the naming as it is already used by some internal code. Will change this accordingly afterward!

**Pull request closed without merging**